### PR TITLE
Add "redundant" hdf5 dependency

### PR DESCRIPTION
**Pull request by xylar**
This is needed for conda-smithy to correctly pin the package.

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -76,6 +78,8 @@ requirements:
     - curl
     - zlib
     - hdf4
+    # without this repeat reference, conda-smithy doesn't pin correctly
+    - hdf5
```
Comment (xylar) [inline recipe/meta.yaml]: Looking at the docs, it looks like the redundancy is only needed in `host`, not `run`.  At the risk of restarting CI, I'm going to fix that.

Comment (beckermr) [reply recipe/meta.yaml]: sure whatever you think is best!

Comment (beckermr) [reply recipe/meta.yaml]: make sure to rerender

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
^^^
Just making sure nothing changes.

File: recipe/meta.yaml (hunk 1 above)
Comment (xylar) [reply recipe/meta.yaml]: Yep, done!

**Comment by xylar** [comment]
Thanks for the review and merge @beckermr

**Pull request merged**